src/Factory/ProductDocumentFactory.php

```diff
@@ -124,9 +125,12 @@ public function createFromSyliusSimpleProductModel(ProductInterface $syliusProdu
         if (null !== $syliusProduct->getMainTaxon()) {
             /** @var TaxonDocument $mainTaxon */
             $mainTaxon = new $this->taxonDocumentClass();
```

Comment:
this naming is confusing.... We should have a `mainTaxon` variable to which we will assign `$syliusProduct->getMainTaxon()`, `$mainTaxonTranslation` as is defined and `$mainTaxonDocument` to which we will assign proper data for indexing